### ISSUE TITLE
no .html suffix of "index" url in generated sw.js 

### ISSUE THREAD

**Issue by levinit** [vite-pwa/sveltekit#22]
![图片](https://user-images.githubusercontent.com/24387319/215427445-cb0a0052-8040-4363-ba46-828af0a61fef.png)

It generated some index.html files, but in sw.js file, all index.html miss `.html` (for example, the highlight one).
In browser, it display an error: 
> sw.js:1  Uncaught (in promise) bad-precaching-response: bad-precaching-response :: [{"url":"https://10.1.1.1/dashboard/index","status":404}]

because the true path is dashboar/index.html.

sveltekitpwa config:

> ```
> SvelteKitPWA({
>       base: '/',
>       scope: '/',
>       srcDir: 'src',
>       strategies: 'injectManifest',
>       filename: 'sw.ts',
>       registerType: 'autoUpdate',
>       // injectRegister: 'inline',
>       injectManifest: {
>         globPatterns: ["client/*.{html,css,js,jpg,png,ico,svg,webp,json,webmanifest}"]
>       },
>       manifest: {
>         name: "app",
>         short_name: "app",
>         description: "app",
>         related_applications: [{
>           "platform": "windows",
>           "url": `${prod_url}/manifest.webmanifest`,
>         }, {
>           "platform": "chrome_web_store",
>           "url": `${prod_url}/manifest.webmanifest`,
>         }],
>         scope: "/",
>         start_url: '/',
>         theme_color: "#93b5cf",
>         background_color: "#ebe7e7",
>         display: 'standalone',
>         icons: [
>           {
>             src: '/pwa-192x192.png',
>             sizes: '192x192',
>             type: 'image/png',
>           },
>           {
>             src: '/pwa-512x512.png',
>             sizes: '512x512',
>             type: 'image/png',
>           },
>           {
>             src: '/pwa-512x512.png',
>             sizes: '512x512',
>             type: 'image/png',
>             purpose: 'any maskable',
>           },
>         ],
>       },
>       devOptions: {
>         enabled: true,
>         type: 'module',
>         navigateFallback: '/'
>       },
>       // if you have shared info in svelte config file put in a separate module and use it also here
>       kit: {}
>     }
> )]
> ```

I remove   `injectManifest:`  configuration, no change for any index url, how to fix it?
Thx!

**Comment by levinit** [comment]
and svelte.config.js

> ```
> import adapter from '@sveltejs/adapter-static';
> import preprocess from 'svelte-preprocess';
> 
> /** @type {import('@sveltejs/kit').Config} */
> const config = {
>     preprocess: preprocess(),
>     kit: {
>         adapter: adapter({
>             fallback: 'index.html'
>         })
>     }
> };
> ```



`src/route/+layout.ts`

> ```
> export const prerender = true;
> export const ssr = false;
> export const trailingSlash = 'always';
> 
> import '../app.css';
> ```

**Comment by levinit** [comment]
I find that in config.ts

```typescript
      if (url.endsWith('.html')) {
        if (url.startsWith('/'))
          url = url.slice(1)

       // url== xxx/index.html , suffix is empty string, so xxx/index.html -> xxx/index
        e.url = url === 'index.html' ? `${base}` : `${base}${url.slice(0, url.lastIndexOf('.'))}${suffix}`
      }
```

**Comment by userquin** [comment]
closed via #32
